### PR TITLE
Security groups create

### PR DESCRIPTION
**Pull request by Joeper214**
Next Goal for #130 

### Goal 
- [x] Create security groups tabpane
- [x] display security group list in tabpane
- [x] Create modal for creating security group
- [x] Saving created security group to backend
- [x] refetch seurity group list and messaging when success or not
- [x] iT18N

### CODE REVIEW

File: app/controllers/nodes_controller.rb [hunk 1]
```diff
@@ -239,15 +239,69 @@ def edit_attributes
...
     rules_summary[:security_groups].map do |item|
       check_socket(item.ip_permissions)
       check_socket(item.ip_permissions_egress)
     end
-
+    sec_groups = File.read("public/security_groups.json")
     @rules_summary = rules_summary[:security_groups]
+    @vpcs = vpcs[:vpcs]
+    @sec_groups = sec_groups
+    puts @sec_groups.to_json
```
Comment (skyarch-networks) [inline app/controllers/nodes_controller.rb]: __[rubocop]__
* Rails/Output: Do not write to stdout. Use Rails' logger if you want to log.

**Comment by Joeper214** [comment]
Please review and Merge PR. @pocke . thanks! :bow:

**Comment by pocke** [comment]
I'll review from now

**Comment by pocke** [comment]
I can click `Security Groups` menu and show security group page when it's disabled.

File: app/controllers/nodes_controller.rb [hunk 2]
```diff
@@ -240,15 +240,26 @@ def edit_attributes
...
+    end
+    vpcs = @infra.ec2.describe_vpcs()
+
 
     rules_summary[:security_groups].map do |item|
       check_socket(item.ip_permissions)
       check_socket(item.ip_permissions_egress)
     end
 
+    sec_groups = File.read("public/security_groups.json")
```
Comment (pocke) [inline app/controllers/nodes_controller.rb]: ?

File: app/views/infrastructures/_security_groups_table.html.erb [hunk 3]
```diff
@@ -0,0 +1,219 @@
+<div class="row">
+  <div class="span*">
```
Comment (pocke) [inline app/views/infrastructures/_security_groups_table.html.erb]: What's `*` ?

File: app/views/infrastructures/modal/_create_security_group_modal.html.erb [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+              <!--<tr>-->
+                <!--<th>Type</th>-->
+                <!--<th>Protocol</th>-->
+                <!--<th>Range</th>-->
+                <!--<th>Source</th>-->
+              <!--</tr>-->
+            <!--</table>-->
+            <!--<button class="btn btn-xs btn-default">Add rule</button>-->
+          <!--</div>-->
+        <!--</div>-->
```
Comment (pocke) [inline app/views/infrastructures/modal/_create_security_group_modal.html.erb]: Is this unneeded ?

File: app/views/infrastructures/_security_groups_table.html.erb (hunk 3 above)
Comment (Joeper214) [inline app/views/infrastructures/_security_groups_table.html.erb]: Sorry. I think it's just mistyped. I'll remove it. thanks!

File: app/controllers/nodes_controller.rb (hunk 2 above)
Comment (Joeper214) [inline app/controllers/nodes_controller.rb]: Sorry that feature is for adding inbound/outbound function. But it's still not done. I'll remove it thanks.

File: app/views/infrastructures/modal/_create_security_group_modal.html.erb (hunk 4 above)
Comment (Joeper214) [inline app/views/infrastructures/modal/_create_security_group_modal.html.erb]: I commented it because that feature is still WIP.

Comment (Joeper214) [inline app/views/infrastructures/modal/_create_security_group_modal.html.erb]: but the Goal is only to create security group. not to add inbound or outbound rules yet. so, I commented it out.

File: app/views/infrastructures/_security_groups_table.html.erb (hunk 3 above)
Comment (pocke) [inline app/views/infrastructures/_security_groups_table.html.erb]: :+1:

**Comment by pocke** [comment]
I cant create new group when `Name` field is empty.

**Comment by pocke** [comment]
rails log

File: config/routes.rb [hunk 5]
```diff
@@ -54,6 +54,7 @@
       get  'run_bootstrap'
       post 'apply_dish'
       post 'submit_groups'
+      post 'create_group'
```
Comment (pocke) [inline config/routes.rb]: `create_group`action doesn't need `:id`.
So, I think `post 'create_group'` should move in `collection`. what do you think?

Comment (Joeper214) [inline config/routes.rb]: :+1:

**Comment by pocke** [comment]
@Joeper214  Can I merge this ?

**Comment by Joeper214** [comment]
If you are done reviewing this PR. Please merge. @pocke :bow:

**Comment by pocke** [comment]
:+1:

**Comment by Joeper214** [comment]
Thanks!. @pocke

**Pull request merged**